File: src/api/employee.ts
import { invoke } from "@tauri-apps/api/core";
import { TEmployee } from "../types";

export const addEmployeeApi = async (data: TEmployee) => {
  try {
    const res = await invoke<TEmployee>("add_employee", { data });
    return res;
  } catch (e) {
    console.error("Error added employee:", e);
  }
};
export const updateEmployeeApi = async (data: TEmployee) => {
  try {
    const res = await invoke<TEmployee>("update_employee_by_id", {
      id: data.id,
      data,
    });
    return res;
  } catch (e) {
    console.error("Error update employee:", e);
  }
};
export const getEmployeesApi = async () => {
  try {
    const res = await invoke<TEmployee[]>("get_employees", {});
    return res;
  } catch (e) {
    console.error("Error get employees:", e);
  }
};
export const deleteEmployeeApi = async (id: string) => {
  try {
    const res = await invoke<TEmployee[]>("delete_employee", { id });
    return res;
  } catch (e) {
    console.error("Error delete employee:", e);
  }
};

File: src-tauri/src/handlers/employee.rs
use crate::datastore::employee;
use crate::types::Employee;

#[tauri::command]
pub async fn add_employee(data: Employee, window: tauri::Window) -> Result<String, String> {
    employee::add_employee_db(data, window)
}

#[tauri::command]
pub async fn update_employee_by_id(
    id: String,
    data: Employee,
    window: tauri::Window,
) -> Result<(), String> {
    employee::update_employee_by_id_db(&id, data, window)
}

#[tauri::command]
pub async fn get_employees(window: tauri::Window) -> Result<Vec<Employee>, String> {
    employee::get_employees_db(window)
}

#[tauri::command]
pub async fn get_employee_by_id(id: String, window: tauri::Window) -> Result<Employee, String> {
    employee::get_employee_by_id_db(&id, window)
}

#[tauri::command]
pub async fn delete_employee(id: String, window: tauri::Window) -> Result<(), String> {
    employee::delete_employee_db(&id, window)
}
